### ISSUE TITLE
on click of refresh event

### ISSUE THREAD

**Issue by rajeshbtlit** [wenzhixin/bootstrap-table#1838]
Hi, We need to handle the onclick of refresh event, could you please let us know how to handle this , for example, when user clickc on refresh button on table,  
`
$('#tableInstance').bootstrapTable({
}).on('refresh', function (e, data) {
//some logic here
   })`

**Comment by dabros** [comment]
@rajeshbtlit - there are ample callbacks already, `onPostBody` seems appropriate, or even `onLoadSuccess`. There is an `onRefreshOptions`, but i couldnt say whether that would fit your use case.

Have a look at how http://issues.wenzhixin.net.cn/bootstrap-table/index.html logs callbacks and play till you find one that fits.

Also, keep in mind that its hard to think of any code needing to be executed ONLY on 'refresh', but NOT on any other update of table or content (`onPostBody`). If there is, then maybe it should not be called using base code callback, but rather in whatever handler/function you have that calls that `refresh`, perhaps via toggle variable (simple bool that if true triggers extra code inside `onPostBody` or similar callback before reverting to false till next needed)

**Comment by rajeshbtlit** [comment]
Thanks dabros, much appreciated, i have a custom search option , where the search would be on server side than on the client side, so i have added a custom search input text box. After the search, the table gets updated with searched result. But on refresh the request is firing to the same searched url, rather than refreshing the table. Somewhere its caching the previous searched url. If i refresh the page, then the complete table is refreshed.

So the reason i asked for onclick of refresh was to use this 

`//refresh the event table
 $('#tableEventCatalog').bootstrapTable('refresh', {
       url: 'wf/Admin/eventcatalog'
   });`

Thanks,
Rajesh

**Comment by dabros** [comment]
@rajeshbtlit - that makes me think the issue is param not passing through to data-url?

http://issues.wenzhixin.net.cn/bootstrap-table/#options/custom-toolbar.html

That example is the closest to what i think your trying to do, and shows how to use `queryParam` and such to pass the search term through.

To help you more though, we really need a fiddle as this doesnt sound like callback issue, but a problem with how you set it up and/or your data source maybe even.

See the data-url used in that custom toolbar example, and see the fiddles here: https://github.com/wenzhixin/bootstrap-table/blob/master/CONTRIBUTING.md#bug-reports, then please provide the closest to yours as possible, actual data-url can be the demo if needed.

**Comment by rajeshbtlit** [comment]
Guys, could you please help here, i need to handle on click of refresh event.
@djhvscf , could you please share your thoughts, I am grateful to you.

**Comment by dabros** [comment]
@rajeshbtlit - you still not providing a fiddle https://github.com/wenzhixin/bootstrap-table/blob/master/CONTRIBUTING.md#bug-reports or more details on how we can replicate, as I just cant see this issue and have pointed you to working examples.

Regarding events in themselves, all event callbacks are detailed here: http://bootstrap-table.wenzhixin.net.cn/documentation/#events

Also, if you hit F12 and watch console then you can use http://issues.wenzhixin.net.cn/bootstrap-table/ to get better idea when some events trigger.

But `onLoadSuccess` or `onPostBody` or `onPostHeader` (with possible use of simple bool flag variables to avoid triggering too many times) should answer almost any use case. You can set that to `true` wherever you call refresh from, then back to `false` after extra logic complete.

Though we still need fiddle or more step-by-step clarification on how you seeing this issue

**Comment by djhvscf** [comment]
@rajeshbtlit see #2106

**Comment by rajeshbtlit** [comment]
Thanks @djhvscf , really appreciate your efforts, I am grateful to you.
I have pulled in these changes and yes i can see the on click of refresh.

The actual problem which i was facing was, i have used a custom search box, which searches from server rather than client side. So on enter of search box I do 

`						$("#searchInstances").keypress(function(event) {
						    if (event.which == 13) {
						        event.preventDefault();
						        var searchInput =$("#searchInstances").val();
						        var searchURL ="wf/Admin/simplifiedInstances?search="+searchInput;
						        //Request the server to fetch the matching records
						        $('#tableInstance').bootstrapTable('refresh', {
				                    url: searchURL
				                }); 
						        
						    }
						});`

Please note the bootstrap table url, its having search parameters with it. 
After search is done and i see the searched results on table.

Now if i click on refresh icon on table toolbar, on Chrome network monitor I can see the table is firing to the last search url only instead of the `data-url="wf/Admin/simplifiedInstances` which i have used for table in html. So i believe the table is caching the url,


So thats the reason I asked the on click of refresh so that i can fire to `$('#tableInstance').bootstrapTable({}).on('refresh.bs.table', function (e, row, $element) {
							$('#tableInstance').bootstrapTable('refresh', {
					                    url: "wf/Admin/simplifiedInstances"
					                });	
						 });`


But now i see that in most of the places I refresh the table with new results and every time this gets called, causing more than 1000 requests on to server.


I am sorry if I am not able to understand you, thing is I cant provide a fiddle for this since search is doing from server side.

In Short, all i want to convey is on click of refresh icon on table toolbar, it should always call the `data-url="wf/Admin/simplifiedInstances` which is used while defining table. Instead of last used url.

Thanks.

**Comment by djhvscf** [comment]
Sure.. I think I know what the problem is now

**Comment by rajeshbtlit** [comment]
Thanks @djhvscf , hopefully you would have a fix.
Once again thanks a lot for your kind help.

**Comment by djhvscf** [comment]
Let me explain me.. You want to click the refresh button but using the initial URL, you don't want to use the URL used in the refresh method, right?

**Comment by rajeshbtlit** [comment]
right, so i have created the table as below:
` <table id="tableInstance" data-toggle="table"
                       data-url="wf/Admin/simplifiedInstances" data-cache="false"
                       data-height="600" data-striped="true" data-pagination="true"
                       data-sort-order="desc" data-sort-name="createTime"
                       data-show-columns="true" data-page-size="50"
                       data-show-multi-sort="true" data-show-refresh="true"
                       data-sort-priority='[{"sortName":"createTime","sortOrder":"desc"},{"sortName":"createdBy","sortOrder":"asc"}]'
                       data-page-list='[50, 75, 100, 200, 300, 500]'>`
followed by columns declaration, so on click of refresh icon on table toolbar, it should fire to `data-url="wf/Admin/simplifiedInstances`  always.

Hope I am able to understand.

Thanks

**Comment by djhvscf** [comment]
I'm going to open this issue and then open a PR where it will be fixed

**Comment by rajeshbtlit** [comment]
hi @djhvscf  I am grateful to you , hope you have planned for this.

**Comment by rajeshbtlit** [comment]
Hi @djhvscf , i am hopeful you would give us some fix . We have a release for our product.
Thanks in Advance.

**Comment by djhvscf** [comment]
See this #2180

**Comment by rajeshbtlit** [comment]
Thanks a lot for your support @djhvscf , am grateful to you.
